pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/parser/ModelParser.groovy

```diff
@@ -581,6 +591,7 @@ class ModelParser implements Parser {
                             } else {
                                 eachStatement(parallelStmt.body.code) {
                                     ModelASTStage parallelStage = parseStage(it)
+                                    parallelStage.failFast = failFast
```

Comment:
Make sure to pass `failFast` to the `parseStages` call for sequential stages below. And we need to update `ast-schema.json` and `JSONParser` too, of course. =)